### PR TITLE
Fixing types & making activities runable

### PR DESCRIPTION
**Pull request by RiskyMH**
this is basicly #5 and possibly #4

### CODE REVIEW

**Review by MaskDuck — DISMISSED** [review]
looks fine

File: src/models/basecog.py [hunk 1]
```diff
@@ -1,8 +1,9 @@
 from nextcord.ext import commands
+from nextcord import Interaction, Client
 
 
 class BaseCog(commands.Cog):
-    def __init__(self, bot):
+    def __init__(self, bot: Client):
```
Comment (MaskDuck) [inline src/models/basecog.py:6]: what exactly is the reason for typehinting with `Client`?

File: src/exts/utils.py [hunk 2]
```diff
@@ -1,38 +1,43 @@
 from models.basecog import BaseCog
-from nextcord import slash_command, SlashOption, Embed
+from nextcord import slash_command, SlashOption, Embed, Interaction, Client
 from dns import resolver
+
+
 class Utils(BaseCog):
-    def __init__(self, bot):
+    def __init__(self, bot: Client):
```
Comment (MaskDuck) [inline src/exts/utils.py:7]: what exactly is the reason to typehint with `Client`?

File: src/exts/suggestion.py [hunk 3]
```diff
@@ -10,21 +10,21 @@
 
 
 class Suggestion(BaseCog):
-    def __init__(self, bot):
+    def __init__(self, bot: Client):
```
Comment (MaskDuck) [inline src/exts/suggestion.py:13]: above

File: src/exts/ping.py [hunk 4]
```diff
@@ -1,13 +1,13 @@
 from models.basecog import BaseCog
-from nextcord import slash_command
+from nextcord import Interaction, slash_command, Client
 
 
 class Ping(BaseCog):
-    def __init__(self, bot):
+    def __init__(self, bot: Client):
```
Comment (MaskDuck) [inline src/exts/ping.py:6]: above

File: src/exts/meta.py [hunk 5]
```diff
@@ -1,17 +1,17 @@
 from models.basecog import BaseCog
-from nextcord import slash_command
+from nextcord import slash_command, Interaction, Client
 
 
 class Meta(BaseCog):
-    def __init__(self, bot):
+    def __init__(self, bot: Client):
```
Comment (MaskDuck) [inline src/exts/meta.py:6]: above

File: src/exts/activity.py [hunk 6]
```diff
@@ -1,10 +1,18 @@
...
+    Interaction,
+    ChannelType,
+    Client,
+)
+from nextcord.abc import GuildChannel
 
 
 class Activity(BaseCog):
-    def __init__(self, bot):
+    def __init__(self, bot: Client):
```
Comment (MaskDuck) [inline src/exts/activity.py:15]: above

**Review by MaskDuck — CHANGES_REQUESTED** [review]

File: src/models/basecog.py (hunk 1 above)
Comment (RiskyMH) [reply src/models/basecog.py:6]: looking at the code, it seemed to produce the most similer typhinting

Comment (MaskDuck) [reply src/models/basecog.py:6]: can you link some snippet?

Comment (RiskyMH) [reply src/models/basecog.py:6]: https://github.dev/nextcord/nextcord/blob/c1c9921cf4a302d51d9bb4657892efa7f7d4ca46/nextcord/ext/commands/bot.py#L1217

I seem to have found the correct class, but the `commands.Bot` seemes to be more adequate, will change accordingly.

Comment (MaskDuck) [reply src/models/basecog.py]: https://github.com/nextcord/nextcord/blob/c1c9921cf4a302d51d9bb4657892efa7f7d4ca46/nextcord/ext/commands/bot.py#L1217 that technique used in that line wasn't typehinting.

Comment (RiskyMH) [reply src/models/basecog.py]: it was extending the class of `nextcord.Client` thus having those typehinting of client, but prob lookes better.

> This class is a subclass of :class:`nextcord.Client` and as a result
    anything that you can do with a :class:`nextcord.Client` you can do with
    this bot.

Comment (MaskDuck) [reply src/models/basecog.py]: ic. still fine because typehinting does nothing at runtime. like you can do `a: int = "1"` and the code still works

**Review by MaskDuck — APPROVED** [review]
good, fine

**Comment by RiskyMH** [comment]
nice

**Pull request merged**